src/realtime/games/BangBang/BangBang.test.ts

```diff
@@ -0,0 +1,232 @@
+import Store from '../../store';
+import { Server } from 'socket.io';
+import { BangBang } from './BangBang';
+
+describe('BangBang Class', () => {
+  const io = new Server();
+  const testID = '1234';
+
+  beforeAll(() => {
+    Store.getInstance();
+    const activeRooms = Store.getInstance().rooms;
+    const newRoomCode = 'ABCD';
+    activeRooms.set(newRoomCode, {
+      ...Store.emptyRoom(),
+      players: [
+        {
+          playerID: 1234,
+          roomCode: 'ABCD',
+          currentlyPlaying: false,
+          nickname: 'Fred',
+          avatarSeed: 'ZXCV',
+          beers: 0,
+          socketID: testID,
+          currentTurn: false,
+        },
+      ],
+    });
+  });
+
+  afterEach(() => {
+    jest.restoreAllMocks();
+    const testRoom = Store.getInstance().rooms.get('ABCD');
+    if (!testRoom) return;
+    testRoom.currentGame = null;
+  });
+
+  describe('Constructor', () => {
+    it('should return gameType as "round"', () => {
+      const bangBangInstance = new BangBang(io, 'ABCD');
+      expect(bangBangInstance.gameType).toEqual('round');
+    });
+
+    it('should return gameName as Bang Bang', () => {
+      const bangBangInstance = new BangBang(io, 'ABCD');
+      expect(bangBangInstance.gameName).toEqual('Bang Bang');
+    });
+
+    it('should initiate playerGameData as empty array', () => {
+      const bangBangInstance = new BangBang(io, 'ABCD');
+      expect(bangBangInstance.playerGameData.length).toEqual(0);
+    });
+  });
+
+  describe('handleMessage method', () => {
+    it('should call beginShooting when moving from cover to game', () => {
+      const bangBangInstance = new BangBang(io, 'testRoom');
```

Comment:
uma pergunta: todas essas declarações de instâncias de BangBang não poderia ser resumida só em uma no início do teste na parte beforeAll?

Reply:
Em alguns casos sim, em outros não, como nos casos que eu mudei dados internos e depois precisava que voltasse pro default (tipo se alguém já atirou ou já votou, ou se uma função preenche algum vetor). Achei mais fácil criar sempre uma instância do zero do que limpar esses valores toda vez e me perder. Posso pesquisar se isso é uma boa prática e caso não seja eu corrijo 🙂